output/output.go

```diff
@@ -1,24 +1,36 @@
 package output
 
-import (
-	"encoding/json"
-	"fmt"
-
-	"github.com/codegangsta/cli"
-)
-
-// JSON prints results in JSON format.
-func JSON(i interface{}) {
-	j, _ := json.Marshal(i)
-	fmt.Println(string(j))
-}
+import "github.com/codegangsta/cli"
 
 // Print prints the results of the CLI command.
-func Print(c *cli.Context, i interface{}, table func(*cli.Context, interface{})) {
+func Print(c *cli.Context, f *func() interface{}, keys []string) {
+	i := (*f)()
 	if c.IsSet("json") {
-		JSON(i)
+		jsonOut(i)
+		return
+	}
+	if len(keys) == 0 {
+		(*f)()
```

Comment:
Automatically call the function if the keys are empty?

Can you document this part, it's a little confusing without context (as to why).